main.go

```diff
@@ -1,12 +1,95 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"encoding/base64"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
 
+	"github.com/ollama/ollama/auth"
 	"github.com/ollama/ollama/cmd"
-	"github.com/spf13/cobra"
+	"github.com/pkg/browser"
 )
 
 func main() {
-	cobra.CheckErr(cmd.NewCLI().ExecuteContext(context.Background()))
+	if err := cmd.NewCLI().ExecuteContext(context.Background()); err != nil {
+		fmt.Fprintln(os.Stderr, "Error:", err)
+
+		if strings.Contains(err.Error(), "unknown ollama key") {
```

Comment:
Can we return a sentinel error here instead of restoring to brittle string checks?

Reply:
This one is a bit tough, because it is across the cli/ollama boundary. I created a shared error type so we can have a constant error message to check for, but a future change could add a `type` to the json of the error response which lets us handle errors better.